### PR TITLE
Olympia fix nft issuance disabled

### LINKED ISSUE #3286: Disable NFT issuance
Make it impossible to issue an NFT, this should be both during video creation, or as a separate step, it should just not be possible. All other extrinsics realted to NFTs do not need to be altered, as they all assume an NFT exists.

This is to limit the likelihood of having to do migrations when we do certain changes to how NFTs work on the runtime side in the near future: https://github.com/Joystream/joystream/issues/3285

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201709380879215/1201866718179361) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by dobertRowneySr**
fixes #3286
~~to be rebased upon #3281 being merged~~ (This PR is disabling nft issuance and nothing else, all other nft fixes are targeting `rhodes`)

### CODE REVIEW

**Comment by ondratra** [comment]
Integration tests for NFT features will need to be disabled as well. 
Calling `nftAuctionAndOffers` needs to be commented out in `tests/integration-tests/src/scenarios/full.ts` and `tests/integration-tests/src/scenarios/content-directory.ts`.

Also, I see some changes to [runtime-modules/content/src/nft/mod.rs](https://github.com/Joystream/joystream/pull/3292/commits/5f95f2240ff7cdfef70d92a3761cb441d1dad6a8#diff-51426774f9053502fb0a278f9f48b7e7936fd43713536ff6585757a436b04987) and [runtime-modules/content/src/lib.rs](https://github.com/Joystream/joystream/pull/3292/commits/5f95f2240ff7cdfef70d92a3761cb441d1dad6a8#diff-530b6b5839302012b1dd920e330e8aed807c41206038f85b19bdf263ace7f349). Are those changes related (only) to disabling NFT issuance? I imagine we will fork `olympia` branch into `rhodes` after we merge this PR. Then we can simply revert this PR in `rhodes` and continue with NFT work there. Because of that, it would be best to limit changes in this PR only to disabling NFTs.

File: runtime-modules/content/src/lib.rs [hunk 1]
```diff
@@ -710,22 +710,23 @@ decl_module! {
...
+            // )?;
 
             // create the video struct
             let video: Video<T> = VideoRecord {
                 in_channel: channel_id,
                 is_censored: false,
                 enable_comments: params.enable_comments,
                 video_post_id:  None,
-                nft_status,
+                nft_status: None,
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs:728]: Please, add the TODO comment here as well.

File: runtime-modules/content/src/lib.rs [hunk 2]
```diff
@@ -1321,6 +1324,8 @@ decl_module! {
             video_id: T::VideoId,
             params: NftIssuanceParameters<T>
         ) {
+            // TODO AUDIT3: remove after open auction fix
+            unimplemented!();
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/lib.rs]: Please, change to the `DispatchError::Other`

File: runtime-modules/content/src/tests/mod.rs [hunk 3]
```diff
@@ -5,6 +5,6 @@ mod curators;
 mod fixtures;
 mod merkle;
 mod mock;
-mod nft;
+//mod nft;
```
Comment (shamil-gadelshin) [inline runtime-modules/content/src/tests/mod.rs:9]: Please, add TODO comment here.

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good. Just a couple of minor issues left.

File: runtime-modules/content/src/lib.rs (hunk 1 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs:728]: fixed in 895e74424d8376ed925328ab1e416678475e1211

File: runtime-modules/content/src/lib.rs (hunk 2 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/lib.rs]: fixed in 895e74424d8376ed925328ab1e416678475e1211

File: runtime-modules/content/src/tests/mod.rs (hunk 3 above)
Comment (dobertRowneySr) [reply runtime-modules/content/src/tests/mod.rs:9]: fixed in 895e74424d8376ed925328ab1e416678475e1211

**Comment by dobertRowneySr** [comment]
> Integration tests for NFT features will need to be disabled as well. Calling `nftAuctionAndOffers` needs to be commented out in `tests/integration-tests/src/scenarios/full.ts` and `tests/integration-tests/src/scenarios/content-directory.ts`.
> 
> Also, I see some changes to [runtime-modules/content/src/nft/mod.rs](https://github.com/Joystream/joystream/pull/3292/commits/5f95f2240ff7cdfef70d92a3761cb441d1dad6a8#diff-51426774f9053502fb0a278f9f48b7e7936fd43713536ff6585757a436b04987) and [runtime-modules/content/src/lib.rs](https://github.com/Joystream/joystream/pull/3292/commits/5f95f2240ff7cdfef70d92a3761cb441d1dad6a8#diff-530b6b5839302012b1dd920e330e8aed807c41206038f85b19bdf263ace7f349). Are those changes related (only) to disabling NFT issuance? I imagine we will fork `olympia` branch into `rhodes` after we merge this PR. Then we can simply revert this PR in `rhodes` and continue with NFT work there. Because of that, it would be best to limit changes in this PR only to disabling NFTs.

The commit history was a bit messy due to the latest decision of disable nft.
But the disabling part of the code are extremely simple and easy to revert

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Tests began to fail. Please, fix it. Also, pay attention to the dead code found by the compiler. I suggest introducing the `allow dead code` directive with the same comment `TODO Rhodes`.

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Pull request merged**